AmplifyPlugins/Auth/AWSAuthPlugin/Support/Utils/AuthErrorHelper.swift

```diff
@@ -0,0 +1,174 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import AWSMobileClient
+
+struct AuthErrorHelper {
+
+    // swiftlint:disable cyclomatic_complexity
+    // swiftlint:disable function_body_length
+    static func map(awsMobileClientError: AWSMobileClientError) -> AmplifyAuthError {
```

Comment:
nit: `map` is usually a term of art used for transforming values in a collection. For one-off transformations, maybe something like 'toAmplifyAuthError()` or similar? Or even constructors like `init(awsMobileClientError: AWSMobileClientError)` and `init(_ error: Error)`? I'm not sure how often we provide initializers for our error cases, though, so aim for consistency with the rest of the codebase here.